sdk/typescript/src/types/objects.ts

```diff
@@ -70,30 +76,122 @@ export const SuiMovePackage = object({
 });
 export type SuiMovePackage = Infer<typeof SuiMovePackage>;
 
-export const SuiData = union([
+export const SuiParsedData = union([
   assign(SuiMoveObject, object({ dataType: literal('moveObject') })),
   assign(SuiMovePackage, object({ dataType: literal('package') })),
 ]);
-export type SuiData = Infer<typeof SuiData>;
+export type SuiParsedData = Infer<typeof SuiParsedData>;
+
+export const SuiRawMoveObject = object({
+  /** Move type (e.g., "0x2::coin::Coin<0x2::sui::SUI>") */
+  type: string(),
+  hasPublicTransfer: boolean(),
+  version: SequenceNumber,
+  bcsBytes: array(number()),
+});
+export type SuiRawMoveObject = Infer<typeof SuiRawMoveObject>;
+
+export const SuiRawMovePackage = object({
+  id: ObjectId,
+  /** A mapping from module name to Move bytecode enocded in base64*/
+  moduleMap: map(string(), string()),
+});
+export type SuiRawMovePackage = Infer<typeof SuiRawMovePackage>;
+
+// TODO(chris): consolidate SuiRawParsedData and SuiRawObject using generics
+export const SuiRawData = union([
+  assign(SuiMoveObject, object({ dataType: literal('moveObject') })),
+  assign(SuiRawMovePackage, object({ dataType: literal('package') })),
+]);
+export type SuiRawData = Infer<typeof SuiRawData>;
 
 export const MIST_PER_SUI = BigInt(1000000000);
 
-export const SuiObject = object({
-  /** The meat of the object */
-  data: SuiData,
-  /** The owner of the object */
-  owner: ObjectOwner,
-  /** The digest of the transaction that created or last mutated this object */
-  previousTransaction: TransactionDigest,
+export const ObjectDigest = string();
+export type ObjectDigest = Infer<typeof ObjectDigest>;
+
+export const SuiObjectData = object({
```

Comment:
We can probably make this a generic so that the return type from GetObject is automatically typed based on the request options. Not something we need to do now but something for the future.

Reply:
Yep I have added this as a TODO comment on line 101